core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -289,64 +377,153 @@ Path versionHintFile() {
     return metadataPath(Util.VERSION_HINT_FILENAME);
   }
 
-  private void writeVersionHint(int versionToWrite) {
+  @VisibleForTesting
+  boolean writeVersionHint(FileSystem fs, Integer versionToWrite) throws Exception {
```

Comment:
Why are we passing in the file system here?

Reply:
I'm just trying to reduce the number of FileSystem objects.